sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
```

Comment:
We need `Datum.IsMax` set appropriately in order to detect when `Datum.Next` would be invalid. We don't have an easy way of computing the actual max date value, so we're just picking an arbitrary one. But you're correct that we should probably enforce that maximum during `Parse{Date,Timestamp}`. And like arithmetic operations on other types, we'll need to check for overflow/underflow (there is a TODO for that, but no issue).